staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/aggregator.go

```diff
@@ -0,0 +1,205 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openapi
```

Comment:
what is this file? Why does it copy tons of code. We can't do that.

Reply:
in apiextensions-apiserver, we need a spec aggregator to 1. keep a {CRD GVK -> spec} mapping 2. merge the specs 3. serve the merged spec on openapi handler (note that it's the openapi handler of apiextensions-apiserver, not kube-aggregator)

so we will have two spec aggregators:
- one in kube-aggregator that keeps {1 aggregated apiserver -> N apiservices -> 1 spec} mapping
- one in apiextensions-apiserver that keeps {1 CRD GVK -> 1 spec} mapping

would like to share some code and have a common aggregator package (apiextensions-apiserver cannot import kube-aggregator directly)